source/material_model/visco_plastic.cc

```diff
@@ -1124,6 +1137,10 @@ namespace aspect
 
           // Limit maximum value of the drucker-prager yield stress
           max_yield_strength = prm.get_double("Maximum yield stress");
+
+          // Include an adiabat temperature gradient in flow laws
+          adiabatic_temperature_gradient_for_viscosity = prm.get_double("Adiabat temperature gradient for viscosity");
+
```

Comment:
Please add the following lines below this line:
```
if (this->get_heating_model_manager().adiabatic_heating_enabled())
AssertThrow (adiabatic_temperature_gradient_for_viscosity == 0.0,
  ExcMessage("If adiabatic heating is enabled you should not add another adiabatic gradient 
                         to the temperature for computing the viscosity, because the ambient 
                         temperature profile already includes the adiabatic gradient.")
```

It will make sure that if you use the extended Boussinesq approximation you are not allowed to set this parameter.

Reply:
Thanks for writing this out for me. I've added it.